core/templates/pages/topic-viewer-page/topic-viewer-page.component.ts

```diff
@@ -16,91 +16,118 @@
-require('services/alerts.service.ts');
-require('services/page-title.service.ts');
-require('services/contextual/url.service.ts');
-require('services/contextual/window-dimensions.service.ts');
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
 
-angular.module('oppia').component('topicViewerPage', {
-  template: require('./topic-viewer-page.component.html'),
-  controller: [
-    '$rootScope', 'AlertsService', 'LoaderService',
-    'PageTitleService', 'TopicViewerBackendApiService',
-    'UrlInterpolationService', 'UrlService', 'WindowDimensionsService',
-    'FATAL_ERROR_CODES',
-    function(
-        $rootScope, AlertsService, LoaderService,
-        PageTitleService, TopicViewerBackendApiService,
-        UrlInterpolationService, UrlService, WindowDimensionsService,
-        FATAL_ERROR_CODES) {
-      var ctrl = this;
-      ctrl.setActiveTab = function(newActiveTabName) {
-        ctrl.activeTab = newActiveTabName;
-      };
-      ctrl.getStaticImageUrl = function(imagePath) {
-        return UrlInterpolationService.getStaticImageUrl(imagePath);
-      };
-      ctrl.checkMobileView = function() {
-        return (WindowDimensionsService.getWidth() < 500);
-      };
-      ctrl.$onInit = function() {
-        ctrl.canonicalStorySummaries = [];
-        if (UrlService.getPathname().endsWith('revision')) {
-          ctrl.setActiveTab('subtopics');
-        } else if (UrlService.getPathname().endsWith('practice')) {
-          ctrl.setActiveTab('practice');
-        } else {
-          ctrl.setActiveTab('story');
-        }
-        ctrl.topicUrlFragment = (
-          UrlService.getTopicUrlFragmentFromLearnerUrl());
-        ctrl.classroomUrlFragment = (
-          UrlService.getClassroomUrlFragmentFromLearnerUrl());
+import { AppConstants } from 'app.constants';
+import { ReadOnlyTopic } from
+  'domain/topic_viewer/read-only-topic-object.factory';
+import { StorySummary } from 'domain/story/StorySummaryObjectFactory';
+import { Subtopic, SkillIdToDescriptionMap } from
+  'domain/topic/SubtopicObjectFactory';
+import { DegreesOfMastery } from
+  'domain/topic_viewer/read-only-topic-object.factory';
+import { TopicViewerBackendApiService } from
+  'domain/topic_viewer/topic-viewer-backend-api.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { AlertsService } from 'services/alerts.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowDimensionsService } from
+  'services/contextual/window-dimensions.service';
+import { LoaderService } from 'services/loader.service';
+import { PageTitleService } from 'services/page-title.service';
+
+
+@Component({
+  selector: 'topic-viewer-page',
+  templateUrl: './topic-viewer-page.component.html',
+  styleUrls: []
+})
+export class TopicViewerPageComponent implements OnInit {
+  activeTab: string = '';
```

Comment:
Do we need types where it can be implied? Having `: string` here makes no difference. Typescript can easily infer the type from the value assigned (which is an empty string). What do you think @kevintab95, @nishantwrp, @bansalnitish, and @seanlip?

Reply:
Sounds good. Created https://github.com/oppia/oppia/issues/10574 and assigned it to me. I'll fix it in a follow-up PR.

Reply:
Can you check with @nishantwrp and @vojtechjelinek? They are the deciders. 

I can see benefits to specifying the type every time (it makes the rule easier for people to follow, you don't have to think about whether to do it or not). But if we have linting in place that reliably catches violations, then that may be a sufficient check too. The important thing is that the "success criteria" are really clearly defined and really easy to evaluate (so, "run this script" or "put types everywhere" works, but not "maybe put a type here if it's implied" where "implied" requires lots of thinking).

Reply:
I agree with @seanlip. We can explicitly define the types of properties of the class. (Similar to the functions).